lib/compat/wordpress-6.0/class-wp-theme-json-gutenberg.php

```diff
@@ -161,4 +214,47 @@ public function get_data() {
 		return $flattened_theme_json;
 	}
 
+	/**
+	 * Given a settings array, it returns the generated rulesets
+	 * for the preset classes.
+	 *
+	 * @param array  $settings Settings to process.
+	 * @param string $selector Selector wrapping the classes.
+	 * @param array  $origins  List of origins to process.
+	 * @return string The result of processing the presets.
+	 */
+	protected static function compute_preset_classes( $settings, $selector, $origins ) {
+		if ( static::ROOT_BLOCK_SELECTOR === $selector ) {
+			// Classes at the global level do not need any CSS prefixed,
+			// and we don't want to increase its specificity.
+			$selector = '';
+		}
+
+		$stylesheet = '';
+		foreach ( static::PRESETS_METADATA as $preset_metadata ) {
+			$slugs = static::get_settings_slugs( $settings, $preset_metadata, $origins );
+			foreach ( $preset_metadata['classes'] as $class => $property ) {
+				foreach ( $slugs as $slug ) {
+					$css_var                 = static::replace_slug_in_string( $preset_metadata['css_vars'], $slug );
+					$class_name              = static::replace_slug_in_string( $class, $slug );
+					$prioritise_user_presets = ( WP_Theme_JSON_Resolver_Gutenberg::theme_has_support() &&
+												isset( $settings['prioritiseUserPresets'] ) &&
+												true === $settings['prioritiseUserPresets'] )
```

Comment:
Though I believe things are spelled correctly here 🇬🇧 , I think the preference is for US spelling: `prioritizeUserPresets`